src/marqo/s2_inference/s2_inference.py

```diff
@@ -47,6 +59,23 @@ def _create_model_cache_key(model_name: str, device: str) -> Tuple:
     model_cache_key = (model_name, device)
     return model_cache_key
 
+
+def _update_model_properties(model: Union[str, dict], model_name: str) -> None:
+    """updates MODEL_PROPERTIES if model details are specified by user
+    """
+    if isinstance(model, dict):
+        model_properties = {model_name: model}
+        MODEL_PROPERTIES['models'].update(model_properties)
+
+
+def _update_available_models(model_cache_key: str, model_name: str, device: str, normalize_embeddings: bool) -> None:
+    """loads the model if it is not already loaded
+    """
+    if model_cache_key not in available_models:
+        available_models[model_cache_key] = _load_model(model_name, device=device)
```

Comment:
this is where it will fail if the model does not exist or there is an error loading